test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+        checker function at start up."""
+        try:
+            async with aiohttp.ClientSession() as s:
+                url = "http://{}:10000/".format(self.cfg["host"])
+                async with s.get(url):
+                    return True
+        except aiohttp.ClientConnectionError:
+            return False
+        # Any other exception may indicate a problem, and is passed to the caller.
+
+    async def start(self) -> None:
+        """Start an installed server. May be used for restarts."""
+        START_TIMEOUT = 300     # seconds
+
+        # Add suite-specific command line options
+        scylla_args = SCYLLA_CMDLINE_OPTIONS + self.cmdline_options
+        env = os.environ.copy()
+        env.clear()     # pass empty env to make user user's SCYLLA_HOME has no impact
+        self.cmd = await asyncio.create_subprocess_exec(
+            self.exe,
+            *scylla_args,
+            cwd=self.cfg["workdir"],
+            stderr=self.log_file,
+            stdout=self.log_file,
+            env=env,
+            preexec_fn=os.setsid,
+        )
+
+        self.start_time = time.time()
+
+        while time.time() < self.start_time + START_TIMEOUT:
+            if self.cmd.returncode:
+                with open(self.log_file_name, 'r') as log_file:
+                    logging.error("failed to start server at host %s", self.cfg["host"])
+                    logging.error("last line of {}:".format(self.log_file_name))
+                    log_file.seek(0, 0)
+                    logging.error(log_file.readlines()[-1].rstrip())
+                    h = logging.getLogger().handlers[0]
+                    logpath = h.baseFilename if hasattr(h, 'baseFilename') else "?"  # type: ignore
+                    raise RuntimeError("""Failed to start server at host {}.
+Check the log files:
+{}
+{}""".format(self.cfg["host"], logpath, self.log_file_name))
+
+            if await self.rest_api_is_up():
+                if await self.cql_is_up():
+                    return
+
+            # Sleep 10 milliseconds and retry
+            await asyncio.sleep(0.1)
+            if self.cfg["seeds"] != self.cfg["host"]:
+                await self.force_schema_migration()
+
+        raise RuntimeError("failed to start server {}, check server log at {}".format(
+            self.host, self.log_file_name))
+
+    async def force_schema_migration(self) -> None:
+        auth = PlainTextAuthProvider(username='cassandra', password='cassandra')
+        with Cluster(contact_points=[self.cfg["seeds"]], auth_provider=auth) as cluster:
+            with cluster.connect() as session:
+                session.execute("CREATE KEYSPACE k WITH REPLICATION = {" +
+                                "'class' : 'SimpleStrategy', 'replication_factor' : 1 }")
+                session.execute("DROP KEYSPACE k")
+
+    async def stop(self) -> None:
+        """Stop a running server. No-op if not running. Uses SIGKILL to
```

Comment:
Why don't we want graceful shutdown? Even if we want "hard kill" for some tests, we still want to test graceful shutdowns as well, no? (so `stop(self)` would be graceful, and we would have a `kill(self)` for hard kill). For graceful shutdown we could use SIGTERM.

Reply:
In 99.9% of the cases where I run a single test function, a graceful shutdown is a waste of time.
If shutdown is very fast (sub-second) and you usually run many tests and not just one (I guess this is typical of test.py) then I guess it won't hurt to have a graceful shutdown.

Reply:
We want both because there will be restarts. I have explained this already in a previous review.